packages/react/leaderboard/src/Leaderboard.tsx

```diff
@@ -279,18 +285,6 @@ export function Leaderboard(props: LeaderboardProps) {
     }
   }, [fetchedData, styles, variant, isStatic, renderChart])
 
-  React.useEffect(() => {
-    try {
-      if (variant !== 'bar' && variant !== 'table') {
-        // console.error('InvalidPropsError: `variant` prop must be either `bar` or `table`') we will set logs as a feature later
-        throw new Error('InvalidPropsError')
-      }
-      setPropsMismatch(false)
-    } catch {
-      setPropsMismatch(true)
-    }
-  }, [variant])
-
```

Comment:
This was causing a bug, when there was a props mismatch error such as the user not passing rows but passing header for example, if the variant was correct this was setting the props mismatch error to `false` anyways, overriding the `true` value from `handlePropsMismatch`. To fix it, I moved this to the `handlePropsMismatch` function.